tests/test_uploads.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+'''Tests for image upload API endpoint.'''
+
+import os.path
+import sys
+import unittest
+
+from flask import Request
+from StringIO import StringIO
+from werkzeug import FileStorage
+from werkzeug.datastructures import MultiDict
+
+# Add webcompat module to import path
+sys.path.append(os.path.realpath(os.pardir))
+
+from webcompat import app
+
+
+class TestUploads(unittest.TestCase):
+    # modified from http://prschmid.blogspot.com/2013/05/unit-testing-flask-file-uploads-without.html
+    def setUp(self):
+        app.config['TESTING'] = True
+        self.app = app
+        self.test_client = self.app.test_client()
+
+    def tearDown(self):
+        pass
+
+    def testGet(self):
+        '''Test that /upload/ doesn't let you GET.'''
+        rv = self.test_client.get('/upload/')
+        self.assertEqual(rv.status_code, 404)
+
+    def testBadUploads(self):
+        # Loop over some files and the status codes that we are expecting
+        for filename, status_code in \
+                (('foo.xxx', 415), ('foo', 415), ('foo.rb', 415)):
+
+            # The reason why we are defining it in here and not outside
+            # this method is that we are setting the filename of the
+            # TestingFileStorage to be the one in the for loop. This way
+            # we can ensure that the filename that we are "uploading"
+            # is the same as the one being used by the application
+            class TestingRequest(Request):
+                """A testing request to use that will return a
+                TestingFileStorage to test the uploading."""
+                @property
+                def files(self):
+                    d = MultiDict()
+                    d['image'] = TestingFileStorage(filename=filename)
+                    return d
+
```

Comment:
Maybe to explain a bit. When you put a `def __init__(self, blah):` in a `class`. the blah will be set each time you initiate the class. On the other hand, if you were just putting a variable under the `class` it will be shared by all the instances.